src/chttpd.erl

```diff
@@ -109,7 +109,8 @@ start_link(https) ->
     start_link(https, Options).
 
 start_link(Name, Options) ->
-    IP = with_default(config:get("chttpd", "bind_address"), any),
+    IP = config:get("chttpd", "bind_address", "any"),
+    couch_httpd:validate_bind_address(IP),
```

Comment:
you need to use the return value of validate_bind_address here (to have it converted to atom any)